[WIP] Add the "testfarm" as a way to instantiante two test pods

cmrd-senya opened the pull request:
This allows to intantiate two pods to test federation on.

"vagrant-group" vagrant plugin is used to group VMs.
The testfarm may be started with:
vagrant group up testfarm

Also requires "vagrant-hosts" plugin.

joebew42 commented on Vagrantfile:
```diff
@@ -28,6 +28,25 @@ Vagrant.configure("2") do |config|
+      # boxes for the test suite. But nothing must change if you disable this and fall back to
+      # "puppetlabs/ubuntu-14.04-64-puppet".
+      dev.vm.box = "diaspora-pod"
+      dev.vm.box_url = "ftp://anonymous:@188.32.218.161/usb1_1/package.box"
```
Why you defined this box url ?

cmrd-senya replied:
The comment two lines above explains it :)

cmrd-senya replied:
It was very useful for my  usecase to speed up the deployment, but pointing to some extra url is not a very good thing. I think I may automate the creation of the speed-up box and make it optional.

Commit:
Add the "testfarm" as a way to instantiante two test pods

This allows to intantiate two pods to test federation on.

"vagrant-group" vagrant plugin is used to group VMs.
The testfarm may be started with:
vagrant group up testfarm

Also requires "vagrant-hosts" plugin.

Commit:
Support LXC based test platform

cmrd-senya commented:
So as you said in #15 
> The capistrano deployment process in development environment is only meant for testing purpose (I want to know if deployment works) and not for development.

Then I suppose I can use the development capistrano stage for my testing purposes here instead of creating a new "test" stage?

The deployment of "development" now works only if ```set :bundle_without, []``` is added to the stage definition, otherwise you get an error because some modules aren't required by bundler.

Commit:
Move "development" to LXC

joebew42 commented:
> Then I suppose I can use the development capistrano stage for my testing purposes here instead of creating a new "test" stage?

If per "testing purposes" do you mean the deploy of branch "develop" in the development environment (development.diaspora.local), yes.

> The deployment of the "development" stage now works only if set :bundle_without, [] is added to the stage definition, otherwise you get an error because some modules aren't required by bundler.

Maybe some last commit in "develop" branch caused this regression. This can be fixed by modifying the `config/deploy/development.rb`. Can you apply these changes ?

cmrd-senya commented:
>Maybe some last commit in "develop" branch of diaspora caused this regression. This can be fixed by modifying the config/deploy/development.rb. Can you apply these changes ?
Yes, I'll test it now to be sure and make a PR then.

cmrd-senya commented:
Ok, I'll close this, since my branch keeps on changing anyway and you don't look interested in this changes in upstream. Also #16 must be fixed first, because I want to use LXC here anyway.

joebew42 commented:
Sorry but I didn't have time to check this pull request. I am interested to provide support for LXC and a test farm. Give me some time to review your work. I think that README.md must be change, also.

cmrd-senya commented:
Ok, thanks. The changes here aren't stabilized though, so I'll set this to [WIP] then. I'll report when it's more or less ready.

LXC must be added in less hacky way than here, but I'm not sure if Vagrant will allow it easily.

Commit:
Add API to use diaspora-replica in automated environment

Commit:
Switch to different branch of "puppet/modules/diaspora"

This branch of "puppet/modules/diaspora" contains version with
proposed allow_username_change feature.
Also this commit updates API.

Commit:
fixup! Switch to different branch of "puppet/modules/diaspora"

Commit:
ruby 2.3.1

Commit:
pod count workaround

Commit:
Don't depend on diaspora_api anymore

also change logger filename pattern
